test/compilable/readmodify_structclass.d

```diff
@@ -0,0 +1,35 @@
+/*
+REQUIRED_ARGS:
+TEST_OUTPUT:
+---
+---
+*/
+shared struct S
+{
+    int x = 0;
+
+	int opUnary(string s)() if (s == "++")
+	{
+		import core.atomic : atomicOp;
+		return atomicOp!"+="(x, 1);
+	}
+}
+
+shared class C
+{
+    int x = 0;
+
+	int opUnary(string s)() if (s == "++")
+	{
+		import core.atomic : atomicOp;
+		return atomicOp!"+="(x, 1);
+	}
+}
+
+void main()
+{
+    S s;
+    s++;
+    shared(C) c = new C();
```

Comment:
Why does this need to be shared if `C` is a `shared class` ?

Reply:
Using shared in the class declaration does not seem to make the class type "implicitly" shared unlike what it does in a struct